Doc/library/inspect.rst

```diff
@@ -550,7 +550,7 @@ The Signature object represents the call signature of a callable object and its
 return annotation.  To retrieve a Signature object, use the :func:`signature`
 function.
 
-.. function:: signature(callable, \*, follow_wrapped=True)
+.. function:: signature(callable_or_text, \*, follow_wrapped=True)
```

Comment:
Sorry, I don't like the `signature()` function to accept both callables and text. This introduces extra complexity to the API for a very narrow use case. 

It also makes the function too dynamic: currently, if a text value is passed to `signature()` by mistake the user would get an exception saying that a callable was expected. With this patch they'll likely get a ParseError.

IMO a public documented `Signature.from_text()` API is enough.